src/core-adapter.js

```diff
@@ -251,7 +251,7 @@ rule GenerateEvents {
   const addNoolsBoilerplate = settingsTasks.isDeclarative && semver.lt(actualCoreVersion, '4.2.0-dev');
   const rules = addNoolsBoilerplate ? addNoolsBoilerplateToCode(settingsTasks.rules) : settingsTasks.rules;
   // do not set the isDeclarative flag when the code has nools boilerplate
-  const rulesAreDeclarative = addNoolsBoilerplate && !!settingsTasks.isDeclarative;
+  const rulesAreDeclarative = !addNoolsBoilerplate && settingsTasks.isDeclarative;
```

Comment:
I think this was an existing bug. :thinking: We want `rulesAreDeclarative` to be `true` when `isDeclarative = true` AND we did NOT add the nools boilerplate right?

Reply:
is there also a bug in the definition of `addNoolsBoilerplate`?  shouldn't we do it if `!settings.isDeclarative || semver.lt(actualCoreVersion, '4.2.0-dev')`?

Reply:
:grimacing: I don't think so... :thinking:   Looking at the [prod code](https://github.com/medic/cht-conf/blob/bf7f647f22f6839ea4d36813fe00e829b6061845/src/fn/upload-app-settings.js#L44) in cht-conf, it looks like we call `nools.addBoilerplateToCode` IF: `appSettings.tasks.isDeclarative && semver.lt(actualCoreVersion, '4.2.0-dev')`  That seems to match what we have here.

The trouble with `rulesAreDeclarative` is that we need to account for the second part of the [logic in cht-conf](https://github.com/medic/cht-conf/blob/bf7f647f22f6839ea4d36813fe00e829b6061845/src/fn/upload-app-settings.js#L47) where, if we add the nools boilerplate, then we never include `isDeclarative`.